full-stack-tests/presentation/src/frontend/UnifiedSelection.test.ts

```diff
@@ -185,6 +188,172 @@ describe("Unified Selection", () => {
       expect(imodel.selectionSet.has(instances.leafElement.key.id)).to.be.true;
     });
 
+    describe("Grouping Node", () => {
+      const addPartition = (classFullName: string, builder: TestIModelBuilder, name: string, parentId = IModel.rootSubjectId) => {
+        const parentProps: RelatedElementProps = {
+          relClassName: "BisCore:SubjectOwnsPartitionElements",
+          id: parentId,
+        };
+
+        const partitionProps: ElementProps = {
+          classFullName,
+          model: IModel.repositoryModelId,
+          parent: parentProps,
+          code: builder.createCode(parentId, BisCodeSpec.informationPartitionElement, name),
+        };
+        return builder.insertElement(partitionProps);
+      };
+
+      const addModel = (classFullName: string, builder: TestIModelBuilder, partitionId: string) => {
+        const modelProps: ModelProps = {
+          modeledElement: new RelatedElement({ id: partitionId }),
+          classFullName,
+          isPrivate: false,
+        };
+        return builder.insertModel(modelProps);
+      };
+
+      const addSpatialCategory = (builder: TestIModelBuilder, modelId: string, name: string, isPrivate?: boolean) => {
+        const spatialCategoryProps: CategoryProps = {
+          classFullName: "BisCore:SpatialCategory",
+          model: IModel.dictionaryId,
+          code: builder.createCode(modelId, BisCodeSpec.spatialCategory, name),
+          isPrivate,
+        };
+        return builder.insertElement(spatialCategoryProps);
+      };
+
+      const addPhysicalObject = (builder: TestIModelBuilder, modelId: string, categoryId: string, elemCode = Code.createEmpty()) => {
+
+        const physicalObjectProps: PhysicalElementProps = {
+          classFullName: "Generic:PhysicalObject",
+          model: modelId,
+          category: categoryId,
+          code: elemCode,
+        };
+        return builder.insertElement(physicalObjectProps);
+      };
+
+      const objectIdArray = Array<Id64String>();
+      let groupingNodesIModel: IModelConnection;
+      beforeEach(async () => {
+        groupingNodesIModel = await buildTestIModel("GroupByClass", (builder) => {
+          const physicalPartitionId = addPartition("BisCore:PhysicalPartition", builder, "TestDrawingModel");
```

Comment:
the name `"TestDrawingModel"` doesn't match type of the model

Reply:
Renamed to TestPhysicalModel